src/migration/v1_11_0.py

```diff
@@ -40,11 +43,93 @@ def monitoring_alert_refactor_alert_number_by_domain_id(mongo_client: MongoCusto
     mongo_client.insert_many('MONITORING', 'alert_number', records, is_new=True)
 
 
+@query
+@check_time
 def monitoring_escalation_policy_change_scope_from_global_to_domain(mongo_client: MongoCustomClient):
     mongo_client.update_many('MONITORING', 'escalation_policy', {"scope": {"$eq": "GLOBAL"}},
                              {"$set": {'scope': 'DOMAIN'}}, upsert=True)
 
 
+@query
+@check_time
+def inventory_cloud_service_refactor_data_structure(mongo_client: MongoCustomClient):
+    projection = {
+        'provider': 1,
+        'metadata': 1,
+        'tags': 1,
+        'collection_info': 1
+    }
+
+    for cloud_services in mongo_client.find_by_pagination('INVENTORY', 'cloud_service', {}, projection):
+
+        operations = []
+        for cloud_service in cloud_services:
+            provider = cloud_service.get('provider', 'custom')
+            metadata = cloud_service.get('metadata', {})
+            tags = cloud_service.get('tags', {})
+            collection_info = cloud_service.get('collection_info', {})
+
+            update_fields = {"$set": {}}
+
+            if tags and isinstance(tags, list):
+                new_tags = {}
+                new_tag_keys = {}
+
+                for tag in tags:
+                    tag_key = str(tag['key'])
+                    tag_value = str(tag['value'])
+                    tag_type = str(tag['type'])
+                    tag_provider = str(tag.get('provider', 'custom'))
+
+                    hash_value = string_to_hash(tag_key)
+
+                    if new_tags.get(tag_provider):
+                        new_tags[tag_provider].update({hash_value: {'key': tag_key, 'value': tag_value}})
+                    else:
+                        new_tags[tag_provider] = {hash_value: {'key': tag_key, 'value': tag_value}}
+
+                    if tag_type == 'PROVIDER' or 'MANAGED':
```

Comment:
change condition
`if tag_type != 'CUSTOM':`

Reply:
have reflected. thank you